### PR TITLE
implement encodingBool

### LINKED ISSUE #84: [encode] Implement encodeBool
**Subject**
encode

**Detail**
```go
// EncodeOperand boolean to hexadecimal bytes
// ex) bool true => 01
// ex) bool false => 00
// TODO: implement w/ test cases :-)
func encodeBool(operand bool) ([]byte, error) {
	return nil, nil
}
```

Implement this with test case!

**Reference**
#29

### PR DESCRIPTION
**Pull request by geekseong**
resolved: #84 

details:
implement encodingBool

1. implement encodingBool
2. write down test case

- [ ] Test case

### CODE REVIEW

File: encoding/encode.go [hunk 1]
```diff
@@ -66,13 +66,29 @@ func encodeInt(operand int) ([]byte, error) {
 // ex) string "abc" => 0x616263
 // TODO: implement w/ test cases :-)
```
Comment (AgwaB) [inline encoding/encode.go]: plz remove TODO comment !

File: encoding/encode.go [hunk 2]
```diff
@@ -66,13 +66,29 @@ func encodeInt(operand int) ([]byte, error) {
...
+		return nil, err
+	}
+
+	return dst, nil
 }
 
 // Encode boolean to hexadecimal bytes
 // ex) bool true => 0x01
 // ex) bool false => 0x00
 // TODO: implement w/ test cases :-)
```
Comment (AgwaB) [inline encoding/encode.go]: here too

**Comment by AgwaB** [comment]
LGTM

File: encoding/encode.go (hunk 1 above)
Comment (geekseong) [reply encoding/encode.go]: i removed it

File: encoding/encode.go (hunk 2 above)
Comment (geekseong) [reply encoding/encode.go]: i removed it

**Comment by zeroFruit** [comment]
How about reset and commit --amend last 2 commits?
> * encodingString bug fix
> * remove TODO comment

File: encoding/encode_internal_test.go [hunk 3]
```diff
@@ -53,3 +53,68 @@ func Test_encodeInt(t *testing.T) {
...
+	for i, test := range tests {
+		op := test.operand
+		bytecode, err := encodeString(op)
+
+		if err != nil {
+			t.Fatalf("test[%d] - encodeInt() had error. err=%v", i, err)
+		}
+
+		if !bytes.Equal(bytecode, test.expected) {
+			t.Errorf("test[%d] - encodeInt() result wrong. expected=%x, got=%x", i, test.expected, bytecode)
```
Comment (boohyunsik) [inline encoding/encode_internal_test.go]: In line 85, you use t.Fatalf. So I think it is better using same method.

File: encoding/encode_internal_test.go [hunk 4]
```diff
@@ -53,3 +53,68 @@ func Test_encodeInt(t *testing.T) {
...
+	for i, test := range tests {
+		op := test.operand
+		bytecode, err := encodeBool(op)
+
+		if err != nil {
+			t.Fatalf("test[%d] - encodeInt() had error. err=%v", i, err)
+		}
+
+		if !bytes.Equal(bytecode, test.expected) {
+			t.Errorf("test[%d] - encodeInt() result wrong. expected=%x, got=%x", i, test.expected, bytecode)
```
Comment (boohyunsik) [inline encoding/encode_internal_test.go]: Same as above comment.

**Comment by junbeomlee** [comment]
Size of the encoded data should be 4bytes!!

**Comment by AgwaB** [comment]
@junbeomlee Are you talking except a string, right?

**Comment by junbeomlee** [comment]
Yes @leesd556

**Comment by geekseong** [comment]
why dose it return 4 bytes?

**Comment by boohyunsik** [comment]
Maybe guys don't like commit message like 'merge...'.
Plz change commit msg :-D

**Review by junbeomlee — APPROVED** [review]
Good work!

**Pull request merged**